test/pokemon-route-test.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+require('./test-env.js');
+
+const expect = require('chai').expect;
+const superagent = require('superagent');
+
+const Trainer = require('../model/trainer.js');
+const Pokemon = require('../model/pokemon.js');
+
+require('../server.js');
+
+const apiURL = `http://localhost:${process.env.PORT}`;
+
+describe('Testing pokemon router', function(){
+  afterEach((done) => {
+    Promise.all([
+      Trainer.remove({}),
+      Pokemon.remove({}),
+    ])
+    .then(() => done())
+    .catch(done);
+  });
+
+  describe('Testing POST with invalid endpoint', function(){
+    it('should return a status code of 404', (done) => {
+      superagent.post(`${apiURL}/api/boooyaa`)
+      .send({})
+      .then(done)
+      .catch((err) => {
+        expect(err.status).to.equal(404);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('Testing POST with bad request', function(){
+    before((done) => {
```

Comment:
No need to go through the trouble of mocking data when you're purposely trying to get an error. You can just send an empty object.